coding_systems/snomedct/tests/test_ecl_parser.py

```diff
@@ -24,6 +26,40 @@ def test_handle(subtests):
             assert handled["excluded"] == excluded
 
 
-def test_parse_error():
+def test_parse_error_syntax_error():
     with pytest.raises(ParseError):
         handle("(111111 OR 222222) MINUS (333333 OR 444444")
+
+
+def test_parse_error_attempting_full_context():
+    with pytest.raises(ParseError, match="Attempting full context"):
+        handle("111111 < 222222")
+
+
+def test_parse_error_ambiguity():
+    with pytest.raises(ParseError, match="Attempting full context"):
+        handle("111111 < 222222")
+
+
+@patch("coding_systems.snomedct.ecl_parser.parse")
+def test_error_unknown_tree_type(mock_parse):
+    bad_tree = {"type": "unk", "operator": None, "value": "111111"}
+    mock_parse.return_value = bad_tree
```

Comment:
I'm not sure what this is really testing.  Are we mocking to ensure full coverage?

In pure computation (eg parsing) if we can't hit a branch without mocking, I'd rather remove the branch, or at least exclude it from coverage.

Reply:
I'll exclude it - I couldn't find a simple test that would return an unexpected tree type, but I don't feel like I understand the ECL enough to be sure that it couldn't happen.